src/dsp_tools/commands/excel2xml/excel2xml_lib.py

```diff
@@ -1883,41 +1811,286 @@ def make_video_segment(  # noqa: D417 (undocumented-param)
+        >>> video_segment.append(excel2xml.make_hasKeyword_prop("keyword of my segment"))
+        >>> video_segment.append(excel2xml.make_hasKeyword_prop("another keyword"))
+        >>> # add other properties
+        >>> root.append(video_segment)
+
+    Returns:
+        an etree._Element that can be appended to an audio/video resource with `segment.append(make_hasKeyword_prop(...))`
+    """
+    if not isinstance(keyword, str) or not check_notna(keyword):
+        msg = (
+            f"Validation Error in resource '{calling_resource}', property 'hasKeyword': "
+            f"The following doesn't seem to be a valid string: '{keyword}'"
+        )
+        warnings.warn(DspToolsUserWarning(msg))
+    prop = etree.Element("{%s}hasKeyword" % xml_namespace_map[None], permissions=permissions)
+    if comment:
+        prop.set("comment", comment)
+    prop.text = keyword
+    return prop
+
+
+def make_hasComment_prop(
+    comment_text: str, permissions: str = "prop-default", comment: str | None = None, calling_resource: str = ""
+) -> etree._Element:
+    """
+    Make a `<hasComment>` property for a `<video-segment>` or `<audio-segment>`.
+
+    Args:
+        comment_text: a text with some background info about the segment. Can be formatted with tags.
+        permissions: defaults to "prop-default".
+        comment: optional comment for this property. Defaults to None.
+        calling_resource: the name of the parent resource (for better error messages)
+
+    Warns:
+        - If the comment text is not a valid string
+
+    Examples:
+        >>> video_segment = excel2xml.make_video_segment("label", "id")
+        >>> video_segment.append(excel2xml.make_hasComment_prop("comment of my segment"))
+        >>> video_segment.append(excel2xml.make_hasComment_prop("another comment"))
+        >>> # add other properties
+        >>> root.append(video_segment)
+
+    Returns:
+        an etree._Element that can be appended to an audio/video resource with `segment.append(make_hasComment_prop(...))`
+    """
+    if not isinstance(comment_text, str) or not check_notna(comment_text):
+        msg = (
+            f"Validation Error in resource '{calling_resource}', property 'hasComment': "
+            f"The following doesn't seem to be a valid string: '{comment_text}'"
+        )
+        warnings.warn(DspToolsUserWarning(msg))
+    prop = etree.Element("{%s}hasComment" % xml_namespace_map[None], permissions=permissions)
+    if comment:
+        prop.set("comment", comment)
+    prop = _add_richtext_to_etree_element(comment_text, prop)
+    return prop
+
+
+def make_hasDescription_prop(
+    description: str, permissions: str = "prop-default", comment: str | None = None, calling_resource: str = ""
+) -> etree._Element:
+    """
+    Make a `<hasDescription>` property for a `<video-segment>` or `<audio-segment>`.
+
+    Args:
+        description: a text with some background info about the segment. Can be formatted with tags.
```

Comment:
Perhaps also mention this in the documentation.

Reply:
Done: https://github.com/dasch-swiss/dsp-tools/commit/7f75622e723d53ed19f503976b5677e7e32547c4